odl/trafos/wavelet.py

```diff
@@ -41,6 +41,114 @@
 _SUPPORTED_IMPL = ('pywt',)
 
 
+def coeff_size_list_axes(shape, wbasis, mode, axes):
```

Comment:
This doesn't have to be a whole new function. You can add an optional `axes` argument to the existing function and do the default case if it's not provided. The easiest way to do it is recursively, something like this:
```python
def coeff_size_list_axes(shape, wbasis, mode, axes=None):
    if axes is None:
        return coeff_size_list_axes(shape, wbasis, mode, [1] * len(shape))

    # Below the case with axes is implemented
```